constraint/pkg/client/constraint_client.go

```diff
@@ -16,24 +16,34 @@ type constraintClient struct {
 	// constraint is a copy of the original Constraint added to Client.
 	constraint *unstructured.Unstructured
 
-	// enforcementAction is what should be done if the Constraint is violated or
-	// fails to run on a review.
-	enforcementAction string
-
 	// matchers are the per-target Matchers for this Constraint.
 	matchers map[string]constraints.Matcher
+
+	// enforcementActionsForEP stores precompiled enforcement actions for each enforcement point.
+	enforcementActionsForEP map[string][]string
 }
 
 func (c *constraintClient) getConstraint() *unstructured.Unstructured {
 	return c.constraint.DeepCopy()
 }
 
-func (c *constraintClient) matches(target string, review interface{}) *constraintMatchResult {
+func (c *constraintClient) matches(target string, review interface{}, sourceEPs ...string) *constraintMatchResult {
 	matcher, found := c.matchers[target]
 	if !found {
 		return nil
 	}
 
+	var actions []string
+	for _, ep := range sourceEPs {
+		if acts, found := c.enforcementActionsForEP[ep]; found {
```

Comment:
check case sensitivity and spaces

Reply:
Storing the initial value in lowercase so this should already be in lowercase.